### PR TITLE
Feat: [IDP-267]: IDP Bug fixes.

### PR DESCRIPTION
**Pull request by sathish-soundarapandian**
## Describe your changes
1. Spell change for harnessCiCdAnnotations
2. Refactor package name from services to service in onboarding module
3. Git optimization - Remove cloneRepoAndCopyToDestDir function call as commitAndPush already does it internally. Minor improvement in timing
4. Tags handling for Org, Project, Service tags. In case of null and empty values, don't set it for backstage entity tags.
5. Account level and Org level services will be added to component entites as of now. But harness CI CD annotations can be only for project level services
6. Bug fix in setting the count of org, project, service that we send to FE during get entities call
7. Move DB save part of catalogConnector entity to the last i.e after initial import of one entity.
8. Handle cases of collision in entity name
9. Git optimization - Use git sync grpc endpoints for git push operation.
10. Truncate entity names so that backstage doesn't error out those entities with name > than 63 characters
11. Remove registration of all location yaml and related stuff
12. Search and Filter by project support in get harness entities flow
13. Separate APIs for count and data fetch

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)

### CODE REVIEW

**Comment by sathish-soundarapandian** [comment]
trigger messagecheck

**Comment by sandeepa-kv-jois** [comment]
> For example CD. Do we need some kind of sign off from them?

This seems to be used in CD and delegate side. We shouldn't be upgrading the versions unless we absolutely need it. Any upgrade in these lib versions would warrant signoff from all teams who are using them.

**Review by sandeepa-kv-jois — COMMENTED** [review]
We shouldn't be upgrading the common lib versions unless we absolutely need it. Any upgrade in these lib versions would warrant signoff from all teams who are using them.

**Review by sathish-soundarapandian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)* status: 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/onboarding/service/impl/OnboardingServiceImpl.java` line 131 at r2](https://reviewable.io/reviews/harness/harness-core/45501#-NRDHOFE2xG2cr9Wxarv:-NRIjFR-DN23z4nBZ-pg:b-f9ow2h) ([raw file](https://github.com/harness/harness-core/blob/565c4fdc3672127ec69e7b857f9e89beeb36cf5b/idp-service/src/main/java/io/harness/idp/onboarding/service/impl/OnboardingServiceImpl.java#L131)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

What did we decide on the paginating the call to ng-manager (in-terms of load/perf)? Did we check with Ankit?
</blockquote></details>

Discussing with Ankit on the same. Will update.

**Comment by sathish-soundarapandian** [comment]
trigger ut3

**Comment by sathish-soundarapandian** [comment]
trigger ut2, ut7

**Comment by sathish-soundarapandian** [comment]
trigger ut2, ut3

**Comment by sathish-soundarapandian** [comment]
trigger ss

**Comment by sathish-soundarapandian** [comment]
trigger ut1, ut2, checkstylecodeformat

**Comment by sathish-soundarapandian** [comment]
trigger codebasehash

**Comment by sathish-soundarapandian** [comment]
trigger ut6, ut7

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -1240,6 +1263,21 @@ components:
...
       required:
         - merged_config
+    GetHarnessEntitiesFilter:
+      title: GetHarnessEntitiesFilter
+      x-stoplight:
+        id: ov49atjlcmu5s
+      type: object
+      description: Get Harness Entities Filter
+      properties:
+        projectsToFilter:
```
Comment (mankrit-singh) [inline idp-service/contracts/openapi/v1/openapi.yaml:1273]: projects_to_filter

**Review by harshjain12 — DISMISSED** [review]

File: idp-service/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (sathish-soundarapandian) [reply idp-service/contracts/openapi/v1/openapi.yaml]: done

**Comment by sathish-soundarapandian** [comment]
trigger ut2

**Review by mankrit-singh — COMMENTED** [review]
> Reviewed 4 of 29 files at r1, 4 of 8 files at r5, 10 of 15 files at r6, 1 of 4 files at r7, 17 of 24 files at r8, 1 of 1 files at r9, 4 of 5 files at r10, 4 of 4 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)* status: 3. unresolved discussions. 0 of 1 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)* status: 2. unresolved discussions.

**Review by sathish-soundarapandian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)* status: 1. unresolved discussions. 

___
*[`project/main_maven_install.json` line 11649 at r2](https://reviewable.io/reviews/harness/harness-core/45501#-NRDJCwJ172Cm0jlLlMh:-NSA3-aX8wc_1xPkHyIQ:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/565c4fdc3672127ec69e7b857f9e89beeb36cf5b/project/main_maven_install.json#L11649)):*
<details><summary><i>Previously, sathish-soundarapandian (Sathish) wrote…</i></summary><blockquote>

@vikyathharekal @sandeepa-kv-jois From the library side, I already checked the release notes of each version from the version which we were using previously, there doesn't seems to be any breaking changes.
We need to upgrade this library because the current version that we use don't support shallow clone options.
Moreover the changes shouldn't affect any of the existing flows because the changes will come into picture only when the corresponding params are set.
@sandeepa-kv-jois Any idea who is the right POC to connect with for these changes ? There is no explicit code owners for this module.
</blockquote></details>

Done

**Review by harshjain12 — APPROVED** [review]

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 4 of 8 files at r5, 10 of 15 files at r6, 1 of 4 files at r7, 17 of 24 files at r8, 1 of 1 files at r9, 4 of 5 files at r10, 4 of 4 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)* status: 45 file(s) to review.

**Review by vikyathharekal — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45501)* status: 45 file(s) to review.

**Pull request merged**